src/swap/SwapScreenV2.tsx

```diff
@@ -0,0 +1,1149 @@
+
+            <View>
+              {showCrossChainSwapNotification && (
+                <View style={styles.crossChainNotificationWrapper}>
+                  <CrossChainIndicator />
+                  <Text style={styles.crossChainNotification}>
+                    {t('swapScreen.crossChainNotification')}
+                  </Text>
+                </View>
+              )}
+              <SwapTransactionDetails
+                feeInfoBottomSheetRef={feeInfoBottomSheetRef}
+                slippageInfoBottomSheetRef={slippageInfoBottomSheetRef}
+                estimatedDurationBottomSheetRef={estimatedDurationBottomSheetRef}
+                slippagePercentage={parsedSlippagePercentage}
+                fromToken={fromToken}
+                toToken={toToken}
+                exchangeRatePrice={quote?.price}
+                exchangeRateInfoBottomSheetRef={exchangeRateInfoBottomSheetRef}
+                swapAmount={processedAmountsFrom.token.bignum ?? undefined}
+                fetchingSwapQuote={shouldShowSkeletons}
+                appFee={appFee}
+                estimatedDurationInSeconds={
+                  quote?.swapType === 'cross-chain' ? quote.estimatedDurationInSeconds : undefined
+                }
+                crossChainFee={crossChainFee}
+                networkFee={networkFee}
+              />
+            </View>
+          </View>
+
+          <View style={[styles.warningsContainer, { display: 'flex', justifyContent: 'flex-end' }]}>
+            {warnings.showCrossChainFeeWarning && (
+              <InLineNotification
+                variant={NotificationVariant.Warning}
+                title={t('swapScreen.crossChainFeeWarning.title', {
+                  tokenSymbol: crossChainFeeCurrency?.symbol,
+                })}
+                description={t('swapScreen.crossChainFeeWarning.body', {
+                  networkName:
+                    NETWORK_NAMES[
+                      crossChainFeeCurrency?.networkId || networkConfig.defaultNetworkId
+                    ],
+                  tokenSymbol: crossChainFeeCurrency?.symbol,
+                  tokenAmount: crossChainFee?.nativeTokenBalanceDeficit.abs().toFormat(),
+                })}
+                style={styles.warning}
+              />
+            )}
+            {warnings.showDecreaseSpendForGasWarning && (
+              <InLineNotification
+                variant={NotificationVariant.Warning}
+                title={t('swapScreen.decreaseSwapAmountForGasWarning.title', {
+                  feeTokenSymbol: feeToken?.symbol,
+                })}
+                description={t('swapScreen.decreaseSwapAmountForGasWarning.body', {
+                  feeTokenSymbol: feeToken?.symbol,
+                })}
+                onPressCta={() => {
+                  if (
+                    !quote ||
+                    quote.preparedTransactions.type !== 'need-decrease-spend-amount-for-gas'
+                  )
+                    return
```

Comment:
why we need this condition? it seems to lead to a confusing ux.

if i correctly understand its behavior:
1) the warning is displayed
2) but the CTA press does nothing

why it does nothing? maybe we shoudn't display the warning in the first place?